Tasks/DropValidatorV0/main.ts

```diff
@@ -0,0 +1,62 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as telemetry from 'azure-pipelines-tasks-utility-common/telemetry';
+import * as fs from 'fs';
+
+async function run() {
+    try {
+        const dropValidatorArgs = ['BuildDropPath', 'ManifestPath', 'OutputPath', 'Verbosity', 'ConfigFilePath', 'RootPathFilter'];
+        let dropValidator = await getDropValidator();
+
+        dropValidatorArgs.forEach(arg => {
+            const inputArgValue = tl.getInput(arg);
+            dropValidator = dropValidator.argIf(inputArgValue != null, '-' + arg).argIf(inputArgValue != null, inputArgValue);
+        });
+
+        const result = await dropValidator.exec();
+
+        if (result === 0) {
+            tl.setResult(tl.TaskResult.Succeeded,  tl.loc('TaskSucceeded'));
+        } else {
+            tl.setResult(tl.TaskResult.Failed, tl.loc('TaskFailed'));
+        }
+
+    } catch (err) {
+        tl.setResult(tl.TaskResult.Failed, err.message);
+    } finally {
+        logTelemetry();
+    }
+}
+
+async function getDropValidator() {
+    try {
+        // Get dotnet tool
+        const dotnetPath = tl.which('dotnet', true);
+        const dotnet = tl.tool(dotnetPath);
+
+        // Install or update the dropvalidator tool globally.
+        dotnet.line('tool update Microsoft.DropValidator --global');
```

Comment:
modifying global machine state in a task is problematic. For one thing, multiple agents running on the same machine is a supported scenario.

Reply:
Downloading tools from the internet is also problematic, because running TFS in a no-internet environment is also a supported scenario.

Reply:
I can definitely try to make the tool install local, but if the ask is to not download anything from the internet, this might a be a problem in the future. The only other way would be to distribute the nupkg or the exe with the task, like how we distribute the 7zip exe. Is there any other way you know that we can implement this?